package/src/main.rs

```diff
@@ -444,33 +444,27 @@ fn fetch_a_scie_project(
         .join("scie-pants")
         .join("dev")
         .join("downloads")
-        .join(project_name)
-        .join(tag);
+        .join(project_name);
     ensure_directory(&cache_dir, false)?;
 
-    let cached_binary = cache_dir.join(&file_name);
+    // We proceed with single-checked locking, contention is not a concern in this build process!
+    // We only care about correctness.
+    let target_dir = cache_dir.join(tag);
+    let lock_file = cache_dir.join(format!("{tag}.lck"));
     let lock_fd = std::fs::OpenOptions::new()
         .create(true)
         .truncate(false)
         .write(true)
-        .open(&cached_binary)
+        .open(&lock_file)
         .map_err(|e| {
             Code::FAILURE.with_message(format!(
                 "Failed to open {path} for locking: {e}",
-                path = cached_binary.display()
+                path = lock_file.display()
             ))
         })?;
     let mut lock = fd_lock::RwLock::new(lock_fd);
     let _write_lock = lock.write();
-    if 0 == std::fs::metadata(&cached_binary)
-        .map_err(|e| {
-            Code::FAILURE.with_message(format!(
-                "Failed to retrieve metadata for {path}: {e}",
-                path = cached_binary.display()
-            ))
-        })?
-        .len()
-    {
+    if !target_dir.exists() {
```

Comment:
What's was the atomicity failure vs the previous commits? I thought based on the previous commits that the `target_dir` is created before this?

Reply:
>  the partial binary would be there

Ahhh, okay, I thought this process was also atomic. That makes sense now.

Reply:
My original thought was not about corruption, but rather just network caching - I didn't think we could half unpack a digest

Reply:
Aha, ok.